src/nemos/solvers.py

```diff
@@ -0,0 +1,726 @@
+        init_params : Any
+            pytree containing the initial parameters.
+            For GLMs it's a tuple of (W, b)
+        args:
+            Positional arguments passed to loss function `fun` and its gradient.
+            For GLMs it is:
+                X :
+                    Input data.
+                y :
+                    Output data.
+
+        init_full_gradient : bool, default False
+            Whether to calculate the full gradient at the initial parameters,
+            assuming that args hold the full data set, and store this gradient in the initial state.
+
+        Returns
+        -------
+        state : SVRGState
+            Initialized optimizer state
+        """
+        # substitute None for prox_lambda
+        return super().init_state(init_params, None, *args, **kwargs)
+
+    @partial(jit, static_argnums=(0,))
+    def update(self, current_params: Any, state: SVRGState, *args, **kwargs):
+        """
+        Perform a single parameter update on the passed data (no random sampling or loops)
+        and increment `state.iter_num`.
+
+        Please note that this is called by `GLM.update`, but repeated calls to `GLM.update`
+        on mini-batches passed to it will not result in running the full (Prox-)SVRG,
+        and parts of the algorithm will have to be implemented outside.
+
+        Parameters
+        ----------
+        current_params : Any
+            Parameters at the end of the previous update, used as the starting point for the current update.
+        state : SVRGState
+            Optimizer state at the end of the previous update.
+            Needs to have the current anchor point (xs) and the gradient at the anchor point (df_xs) already set.
+        args:
+            Positional arguments passed to loss function `fun` and its gradient.
+            For GLMs it is:
+                X :
+                    Input data.
+                y :
+                    Output data.
+
+        Returns
+        -------
+        OptStep
+            xs : Any
+                Parameters after taking one step defined in the inner loop of Prox-SVRG.
+            state : SVRGState
+                Updated state.
+        """
+        # substitute None for prox_lambda
+        return super().update(current_params, state, None, *args, **kwargs)
+
+    @partial(jit, static_argnums=(0,))
+    def run(
+        self,
+        init_params: Any,
+        *args,
+    ) -> OptStep:
+        """
+        Run a whole optimization until convergence or until `maxiter` epochs are reached.
+        Called from `GLM.fit` and assumes that X and y are the full data set.
+
+        Parameters
+        ----------
+        init_params : Any
+            Initial parameters to start from.
+        args:
+            Positional arguments passed to loss function `fun` and its gradient.
+            For GLMs it is:
+                X :
+                    Input data.
+                y :
+                    Output data.
```

Comment:
remember to update the docstrings as in proxSVRG

Reply:
Updated!